### PR TITLE
#159000210 download asset-category with assets count as csv data file

### PR DESCRIPTION
**Pull request by Dheavyman**
#### What does this PR do?
- Download asset categories with their assets count as csv data file
#### Description of Task to be completed?
- A developer should be able to download all asset categories and their asset count as a CSV file
#### How should this be manually tested?
- Create a local copy with the following commands: `git fetch origin ft-download-asset-categories-csv-159000210:ft-download-asset-categories-csv-159000210`, then `git checkout ft-download-asset-categories-csv-159000210`
- install the dependencies with the command: `pipenv install --dev`
- Start the application with the command: `python manage.py runserver`
- Using postman, send a GET request to: `localhost:5000/api/v1/asset-categories/export`
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#159000210](https://www.pivotaltracker.com/story/show/159000210)
#### Postman collection:
- https://www.getpostman.com/collections/2d8c1aa4aad75d3cddb0
#### Screenshots (if appropriate)
screen shot 2018-07-14 at 12 42 02 am
screen shot 2018-07-14 at 12 41 36 am

#### Questions:
- N/A

### CODE REVIEW

File: api/views/asset_category_csv.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@api.route('/asset-categories/export')
+class AssetCategoriesCsv(Resource):
+    """
+    Resource class for asset categories csv data files
+    """
+
+    @token_required
+    def get(self):
+        asset_categories = AssetCategory._query(request.args) if request.args \
```
Comment (Victoradukwu) [inline api/views/asset_category_csv.py]: The line should  be: `asset_categories = AssetCategory._query(request.args).all()`. The check for `request.args` is already handled down the line in the `_query` method

Comment (Dheavyman) [reply api/views/asset_category_csv.py]: Thanks for the reminder, making the changes

Comment (Dheavyman) [reply api/views/asset_category_csv.py]: Done

**Comment by brayoh** [comment]
@Dheavyman You forgot to add the postman collection

**Comment by Dheavyman** [comment]
@brayoh Thanks for the reminder, have added the postman collection link to the PR description.

**Comment by SEUNAGBEYE** [comment]
@Dheavyman I suggest you add a `pipenv install --dev` to the "how should this be manually tested" section so anybody following your instructions will not get an error saying the `flask-excel` module is missing.

File: tests/test_asset_category_csv.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        response = client.get(
+            f'{api_base_url_v1}/asset-categories/export')
+        response_json = json.loads(response.data.decode(CHARSET))
+
+        assert response.status_code == 401
+        assert response_json['status'] == 'error'
+        assert response_json['message'] == 'Bad request. Header does not ' \
+                                           'contain an authorization token.'
+
+    def test_export_asset_category_as_csv_success(
```
Comment (frankip) [inline tests/test_asset_category_csv.py]: nice work on this test

File: api/views/asset_category_csv.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    """
+
+    @token_required
+    def get(self):
+        """Download asset categories with their assets count"""
+
+        asset_categories = AssetCategory._query(request.args).all()
+        column_names = ['name', 'assets_count']
+
+        return excel.make_response_from_query_sets(
```
Comment (frankip) [inline api/views/asset_category_csv.py]: 👍

**Pull request merged**